cmd/clusterctl/client/topology.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+)
+
+// DryRunOptions define options for DryRunTopology.
+type DryRunOptions struct {
+	// Kubeconfig defines the kubeconfig to use for accessing the management cluster. If empty,
+	// default rules for kubeconfig discovery will be used.
+	Kubeconfig Kubeconfig
+
+	// Objs is the list of objects that are input to the dryrun operation.
+	Objs []*unstructured.Unstructured
```

Comment:
@fabriziopandini

Reply:
I would stick to unstructured for now, given that I assume most of the users will read the objects should be applied from files like we are doing

Reply:
Just talked to Yuvaraj about it, our idea was to take a quick look if it's easy to change, otherwise just keep it.

The idea behind it is that client.Object is more flexible for consumers as it could be a concrete API type or an unstructured, while unstructured forces every consumer to convert to unstructured to use the API.

Reply:
I checked and making this change will need some refactor. Will address this in follow up.